codec-socks/src/main/java/io/netty/handler/codec/socks/SocksAuthRequestDecoder.java

```diff
@@ -28,39 +28,38 @@
  */
 public class SocksAuthRequestDecoder extends ReplayingDecoder<State> {
 
-    private SocksSubnegotiationVersion version;
-    private int fieldLength;
-    private String username;
-    private String password;
-    private SocksRequest msg = SocksCommonUtils.UNKNOWN_SOCKS_REQUEST;
-
     public SocksAuthRequestDecoder() {
         super(State.CHECK_PROTOCOL_VERSION);
     }
 
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf byteBuf, List<Object> out) throws Exception {
+        String username = null;
         switch (state()) {
             case CHECK_PROTOCOL_VERSION: {
-                version = SocksSubnegotiationVersion.valueOf(byteBuf.readByte());
+                SocksSubnegotiationVersion version = SocksSubnegotiationVersion.valueOf(byteBuf.readByte());
                 if (version != SocksSubnegotiationVersion.AUTH_PASSWORD) {
+                    out.add(SocksCommonUtils.UNKNOWN_SOCKS_REQUEST);
                     break;
                 }
                 checkpoint(State.READ_USERNAME);
             }
             case READ_USERNAME: {
-                fieldLength = byteBuf.readByte();
+                int fieldLength = byteBuf.readByte();
                 username = SocksCommonUtils.readUsAscii(byteBuf, fieldLength);
                 checkpoint(State.READ_PASSWORD);
             }
             case READ_PASSWORD: {
-                fieldLength = byteBuf.readByte();
-                password = SocksCommonUtils.readUsAscii(byteBuf, fieldLength);
-                msg = new SocksAuthRequest(username, password);
+                int fieldLength = byteBuf.readByte();
+                String password = SocksCommonUtils.readUsAscii(byteBuf, fieldLength);
+                out.add(new SocksAuthRequest(username, password));
```

Comment:
I would like a unit test as part of this PR because behavior is potentially changing. Previously we kept a reference to `username` at the class scope. So if there were multiple decode calls where the first decode call saved the `username`, and the second decode call provided enough data to decode the `password` we would have a coherent `password` and `username`. However the behavior has changed in this PR where this is no longer possible and the entire payload must be present in the same `decode` call or we will behave differently.